src/Nethermind/Nethermind.State/StateProvider.cs

```diff
@@ -808,11 +808,6 @@ public void Reset()
             _currentPosition = Resettable.EmptyPosition;
             Array.Clear(_changes, 0, _changes.Length);
             _needsStateRootUpdate = false;
-
-            if (_codeDb is IReadOnlyDb readOnlyDb)
-            {
-                readOnlyDb.ClearTempChanges();
```

Comment:
are those changes ever cleared? Not a leak?

Reply:
Heck. I don't know.